src/proxy/BaseProxy.ts

```diff
@@ -1,24 +1,31 @@
 import {SerializationService} from '../serialization/SerializationService';
 import {Data} from '../serialization/Data';
+import HazelcastClient = require('../HazelcastClient');
+import ClientMessage = require('../ClientMessage');
 export class BaseProxy {
 
-    protected client: any;
-    private name: string;
-    private serviceName: string;
+    protected client: HazelcastClient;
+    protected name: string;
+    protected serviceName: string;
 
-    //TODO client should be of type {real client type}
-    constructor(client: { serializationService : SerializationService}, serviceName: string, name: string) {
+    constructor(client: HazelcastClient, serviceName: string, name: string) {
         this.client = client;
         this.name = name;
         this.serviceName = serviceName;
     }
 
+    protected invokeWithPartitionId(clientMessage: ClientMessage, key: any): Q.Promise<ClientMessage> {
+        var partitionId: number = Math.abs(this.client.getSerializationService().toData(key).getPartitionHash()) % 271;
```

Comment:
PartitionService should handle this. Maybe just an empty one can be better to start with.